src/components/admin/category-bar/CategoryBar.tsx

```diff
@@ -43,23 +90,43 @@ export const CategoryBar = <T,>({
                     ))}
                 </ContextMenuButton>
             )}
-            {categories.map((category) => {
-                const key = getCategoryKey(category);
-                const name = getCategoryDisplayName(category);
-                const selected = !!selectedCategory && key === getCategoryKey(selectedCategory);
+            {showLeftArrow && (
+                <button
+                    className="category-bar-arrow category-bar-arrow-left"
+                    onClick={() => handleScroll('left')}
+                    type="button"
+                >
+                    <ArrowLeftIcon />
+                </button>
+            )}
+            <div ref={categoriesContainerRef} className="category-bar-categories" onScroll={checkScroll}>
+                {categories.map((category) => {
+                    const key = getCategoryKey(category);
+                    const name = getCategoryDisplayName(category);
+                    const selected = !!selectedCategory && key === getCategoryKey(selectedCategory);
 
-                return (
-                    <button
-                        key={key}
-                        onClick={() => onCategorySelect(category)}
-                        className={classNames('category-bar-button', {
-                            'category-bar-button-selected': selected,
-                        })}
-                    >
-                        {name}
-                    </button>
-                );
-            })}
+                    return (
+                        <button
+                            key={key}
+                            onClick={() => onCategorySelect(category)}
+                            className={classNames('category-bar-button', {
+                                'category-bar-button-selected': selected,
+                            })}
+                        >
+                            {name}
+                        </button>
```

Comment:
Every time `CategoryBar` re-renders, we re-map the entire `categories` array and create a new `onClick` inline function for every single button. This forces the entire list of buttons to re-render.
Extract the `<button>` into a separate `CategoryButton` component and wrapping it in `React.memo`. This new component can manage its own `onClick` callback.
```suggestion
                        <CategoryButton
                            key={key}
                            category={category}
                            name={name}
                            isSelected={selected}
                            onSelect={onCategorySelect}
                        />
```

Reply:
Done